src/plugins/calibration/calibration_impl.cpp

```diff
@@ -128,6 +128,33 @@ void CalibrationImpl::calibrate_magnetometer_async(const CalibrationCallback& ca
         command, std::bind(&CalibrationImpl::command_result_callback, this, _1, _2));
 }
 
+void CalibrationImpl::calibrate_levelhorizon_async(const CalibrationCallback& callback)
+{
+    std::lock_guard<std::mutex> lock(_calibration_mutex);
+
+    if (_parent->is_armed()) {
+        report_failed("System is armed.");
+        return;
+    }
```

Comment:
This is actually not done properly, also for the other calls. We should fix it for another time. We should properly answer using the callback (as below).

Reply:
@julianoes addressed.